### PR TITLE
[OpenSprinkler] Use DS annotations, fix static code analysis findings

### PR DESCRIPTION
**Pull request by wborn**
* Use DS annotations
* Fix static code analysis findings

### CODE REVIEW

File: addons/binding/org.openhab.binding.opensprinkler/README.md [hunk 1]
```diff
@@ -10,8 +10,7 @@ Stations can be controlled to be turned on or off and rain sensor state can be r
 
 ## Discovery
 
-*   OpenSprinkler devices can be manually discovered by sending a request to every IP on the network.
-Discovery needs to be run manually as this is a brute force method of finding devices that can saturate network or device available bandwidth.
+OpenSprinkler devices can be manually discovered by sending a request to every IP on the network. Discovery needs to be run manually as this is a brute force method of finding devices that can saturate network or device available bandwidth.
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.opensprinkler/README.md]: Why this change? If I understood correctly something in MD prefer to have a single sentence on a single line.

Comment (wborn) [reply addons/binding/org.openhab.binding.opensprinkler/README.md]: The SAT "MarkdownCheck" was violated because "The line after a Markdown list must be empty." So I removed the bullet because the whole paragraph showed up as a bullet with text.

See: https://github.com/openhab/openhab2-addons/blob/master/addons/binding/org.openhab.binding.opensprinkler/README.md#discovery

Comment (martinvw) [reply addons/binding/org.openhab.binding.opensprinkler/README.md]: Now I only see the \*, actually I was already a sleep but woke up for the baby-son :-D

Comment (martinvw) [reply addons/binding/org.openhab.binding.opensprinkler/README.md]: @bgilmer77 what would be the ideal here? I would assume two lines but indeed lose the \*

Comment (martinvw) [reply addons/binding/org.openhab.binding.opensprinkler/README.md]: @bgilmer77 ping!

Comment (bgilmer77) [reply addons/binding/org.openhab.binding.opensprinkler/README.md]: Okay - I'm awake now...  The Linter takes a little getting used to.  Generally for plain text we want every sentence on its own line.  Without a blank line between two lines, MD will put both sentences together into one paragraph.  When it comes to bullets, if the bullet point has only one sentence, or a partial sentence, then put it on one line without a period at the end.  If the bullet point has more than one sentence, you have to put them together without a carriage return.  In this case, end the second sentence with a period.

In this case, even thought there are bulleted lists above and below this specific topic, the Discovery section looks just fine as a regular paragraph without the bullet.  Adding a single bullet here would not make sense to me.  That is my opinion.

Comment (martinvw) [reply addons/binding/org.openhab.binding.opensprinkler/README.md]: Okay so the ideal would be both sentences on their own line.

Comment (bgilmer77) [reply addons/binding/org.openhab.binding.opensprinkler/README.md]: Exactly - both sentences on their own line.  No bullet.  No blank line in between the two sentences.

Comment (wborn) [reply addons/binding/org.openhab.binding.opensprinkler/README.md]: I've now placed each sentence on its own line.

Probably these checks are based on this [styleguide](https://github.com/openhab/openhab-docs/blob/gh-pages/styleguide.md)? So the rationale for putting both sentences on their own line is explained [here](http://www.cirosantilli.com/markdown-style-guide/#option-wrap-inner-sentence).

To let contributors know what styling is preferred it would be helpful to mention this in the documentation (e.g. [Developing Bindings](https://docs.openhab.org/developers/development/bindings.html) or [Coding Guidelines](https://docs.openhab.org/developers/development/guidelines.html)) @bgilmer77.

What would be the easiest way for developers to check their markdown files for Linter errors?

Does this styling only apply to openHAB Markdown or does it also apply to Eclipse SmartHome?

Comment (martinvw) [reply addons/binding/org.openhab.binding.opensprinkler/README.md]: @bgilmer77 it would be great if you or anyone else from the docs-team can propose an update for the docs mentioned above to clarify some of the rules regarding markdown.

@kaikreuzer would it apply only to openHAB or should it be changed in ESH first or as well?

Comment (martinvw) [reply addons/binding/org.openhab.binding.opensprinkler/README.md]: 👍 so we should just make sure we use that same page or something similar in openHAB
